### PR TITLE
[Ready] Make synth data generation more customiseable

### PR DESCRIPTION
**Pull request by tripathiarpan20**

### CODE REVIEW

File: validator/utils/synthetic/synthetic_utils.py [hunk 1]
```diff
@@ -119,8 +119,12 @@ async def get_random_image_b64(cache: diskcache.Cache) -> str:
         if random.random() < 0.01:
             cache.delete(key)
         return image_b64
+    
+    possible_dims = [256, 512, 768, 1024]
```
Comment (namoray) [inline validator/utils/synthetic/synthetic_utils.py:123]: 256 is too low here, needs to be 512 min, and we need a value above 1024

File: validator/control_node/src/synthetics/synthetic_generation_funcs.py [hunk 2]
```diff
@@ -221,26 +280,29 @@ async def generate_inpaint_synthetic() -> payload_models.InpaintPayload:
     return payload_models.InpaintPayload(
         prompt=prompt,
         negative_prompt=negative_prompt,
-        cfg_scale=2.0,
+        cfg_scale=random.uniform(cfg_scale_range[0], cfg_scale_range[1]),
```
Comment (namoray) [inline validator/control_node/src/synthetics/synthetic_generation_funcs.py:283]: needs to be 0 or 1 dp

File: validator/control_node/src/synthetics/synthetic_generation_funcs.py [hunk 3]
```diff
@@ -152,17 +197,22 @@ async def generate_chat_synthetic(model: str) -> payload_models.ChatPayload:
 
 async def generate_text_to_image_synthetic(
     model: str,
+    max_prompt_words: int = 20,
+    height_range: tuple[int, int] = (512, 1024),
+    width_range: tuple[int, int] = (512, 1024),
+    cfg_scale_range: tuple[float, float] = (1.0, 5.0),
```
Comment (namoray) [inline validator/control_node/src/synthetics/synthetic_generation_funcs.py:203]: dont put defaults to these, will cause more issues

File: validator/control_node/src/synthetics/synthetic_generation_funcs.py [hunk 4]
```diff
@@ -152,17 +197,22 @@ async def generate_chat_synthetic(model: str) -> payload_models.ChatPayload:
...
+    steps_range: tuple[int, int] = (10, 20),
```
Comment (namoray) [inline validator/control_node/src/synthetics/synthetic_generation_funcs.py:204]: same for here, dont put defaults

File: core/task_config.py [hunk 5]
```diff
@@ -184,7 +223,14 @@ def task_configs_factory() -> dict[str, cmodels.FullTaskConfig]:
...
             synthetic_generation_config=cmodels.SyntheticGenerationConfig(
                 func="generate_image_to_image_synthetic",
-                kwargs={"model": FLUX_SCHNELL_IMAGE_TO_IMAGE},
+                kwargs={
+                    "model": FLUX_SCHNELL_IMAGE_TO_IMAGE,
+                    "max_prompt_words" : 20,
+                    "height_range" : (512, 1024),
+                    "width_range" : (512, 1024),
+                    "cfg_scale_range" : (1.0, 5.0),
+                    "steps_range" : (5, 10)
```
Comment (namoray) [inline core/task_config.py:232]: (2, 7)

File: core/task_config.py [hunk 6]
```diff
@@ -161,7 +193,14 @@ def task_configs_factory() -> dict[str, cmodels.FullTaskConfig]:
...
             synthetic_generation_config=cmodels.SyntheticGenerationConfig(
                 func="generate_text_to_image_synthetic",
-                kwargs={"model": FLUX_SCHNELL_TEXT_TO_IMAGE},
+                kwargs={
+                    "model": FLUX_SCHNELL_TEXT_TO_IMAGE,
+                    "max_prompt_words" : 20,
+                    "height_range" : (512, 1024),
+                    "width_range" : (512, 1024),
+                    "cfg_scale_range" : (1.0, 5.0),
+                    "steps_range" : (5, 10)
```
Comment (namoray) [inline core/task_config.py:202]: (2, 7)

File: validator/control_node/src/synthetics/synthetic_generation_funcs.py [hunk 7]
```diff
@@ -80,11 +107,25 @@ def get_randomly_edited_face_picture_for_avatar() -> str | None:
 
     Hence, we can use a single picture and just edit it to generate 2**(1024*1024) unique images
     """
+
     try:
-        my_boy_postie = _load_postie_to_pil("assets/postie.png")
-    except FileNotFoundError:
-        my_boy_postie = _load_postie_to_pil("validator/control_node/assets/postie.png")
-    return _alter_my_boy_postie(my_boy_postie)
+        dim_range = [400, 512, 600, 768]
```
Comment (namoray) [inline validator/control_node/src/synthetics/synthetic_generation_funcs.py:112]: why dis range?

File: validator/control_node/src/synthetics/synthetic_generation_funcs.py (hunk 2 above)
Comment (tripathiarpan20) [reply validator/control_node/src/synthetics/synthetic_generation_funcs.py:283]: CFG scale should between 0 and 1 for inpaint? (just to confirm)

File: validator/control_node/src/synthetics/synthetic_generation_funcs.py (hunk 7 above)
Comment (tripathiarpan20) [reply validator/control_node/src/synthetics/synthetic_generation_funcs.py:112]: random input dims ranging from low to mid quality

File: core/task_config.py [hunk 8]
```diff
@@ -138,7 +163,14 @@ def task_configs_factory() -> dict[str, cmodels.FullTaskConfig]:
             ),
             synthetic_generation_config=cmodels.SyntheticGenerationConfig(
                 func="generate_image_to_image_synthetic",
-                kwargs={"model": PROTEUS_IMAGE_TO_IMAGE},
+                kwargs={
+                    "model": PROTEUS_IMAGE_TO_IMAGE,
+                    "max_prompt_words" : 20,
+                    "height_range" : (512, 1024),
```
[the remark this answers is not in the document]
Comment (tripathiarpan20) [reply core/task_config.py:169]: for all image to image reqs I presume

File: core/task_config.py [hunk 9]
```diff
@@ -161,7 +193,14 @@ def task_configs_factory() -> dict[str, cmodels.FullTaskConfig]:
             ),
             synthetic_generation_config=cmodels.SyntheticGenerationConfig(
                 func="generate_text_to_image_synthetic",
-                kwargs={"model": FLUX_SCHNELL_TEXT_TO_IMAGE},
+                kwargs={
+                    "model": FLUX_SCHNELL_TEXT_TO_IMAGE,
+                    "max_prompt_words" : 20,
+                    "height_range" : (400, 1600),
+                    "width_range" : (512, 1024),
```
Comment (namoray) [inline core/task_config.py:200]: still abit low - landscape images right?

File: validator/control_node/src/synthetics/synthetic_generation_funcs.py (hunk 2 above)
Comment (namoray) [reply validator/control_node/src/synthetics/synthetic_generation_funcs.py:283]: inpaint is ded actually dw.

But dp == decimal place, for all the cfg's they should be 0 or 1 decimal place only

Comment (namoray) [reply validator/control_node/src/synthetics/synthetic_generation_funcs.py:283]: 1dp is good

**Review by namoray — APPROVED** [review]

File: core/task_config.py (hunk 9 above)
Comment (tripathiarpan20) [reply core/task_config.py:200]: tis possible to get 400x1024 & 512x1600 tho, what range would be prefer?

**Pull request merged**